sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java

```diff
@@ -44,8 +44,9 @@ public PagedIterable(PagedFlux<T> pagedFlux) {
      * It will provide same {@link Stream} of T values from starting if called multiple times.
      * @return {@link Stream} of {@link PagedResponse}
      */
-    public Stream<PagedResponse<T>> streamByPage() {
-        return pagedFlux.byPage().toStream();
+    @SuppressWarnings("unchecked")
+    public <P extends PagedResponse<T>> Stream<P> streamByPage() {
```

Comment:
Update the javadocs also to reflect the changes in return types.

Reply:
updated